mlir/include/mlir/Dialect/EmitC/IR/EmitCTypes.td

```diff
@@ -22,6 +22,12 @@ include "mlir/IR/BuiltinTypeInterfaces.td"
 // EmitC type definitions
 //===----------------------------------------------------------------------===//
 
+def Valid_EmitC_Integer_Type : Type<CPred<"emitc::isValidEmitCIntegerType($_self)">,
```

Comment:
```suggestion
def EmitCIntegerType : Type<CPred<"emitc::isSupportedIntegerType($_self)">,
```

Reply:
I renamed the types. While refactoring it, I noticed that the other types tend to follow the naming convention `EmitC_XType`. Should we go with `EmitC_IntegerType`/`EmitC_FloatType` maybe?

Reply:
Well those types, to be precise `EmitC_ArrayType`, `EmitC_OpaqueType` and `EmitC_PointerType`, are EmitC specific and we have our own implementations for it. In contrast to that, `EmitCIntegerType` as wells as `EmitCFloatType`just refer to a collection of (bultin) types without implementing a new type, similar to

https://github.com/llvm/llvm-project/blob/4462bb58be7d68274fb9b443b7bf4803306df87d/mlir/include/mlir/Dialect/EmitC/IR/EmitC.td#L54-L55

Therefore, I am fine with the slightly differing name scheme. @simon-camp your thoughts?

Reply:
I don't have a strong opinion either.

Reply:
Ok, sounds reasonable, let's go with `EmitCIntegerType`/`EmitCFloatType`.